### PR TITLE
ASoC: SOF: use hex_dump_to_buffer to dump call stack

### PR DESCRIPTION
**Pull request by RanderWang**
It is simpler to use hex_dump_to_buffer than dumping
stack by ourself.

Don't check whether the value of stack_words is multiple of 4
because the value of stack_words is always multiple of 4 in
driver and it makes no sense to dump stack with value of 4*x + y

it fix https://github.com/thesofproject/linux/issues/410

It works on WHL

### CODE REVIEW

**Comment by RanderWang** [comment]
@andy-shev hi andy, could do you help to review this PR? Thanks

**Comment by andy-shev** [comment]
I'm on vacation till January

**Comment by plbossart** [comment]
Thanks @andy-shev, have a great vacation.
@RanderWang please update the PR, I agree with all the comments from Andy.

**Comment by RanderWang** [comment]
update my PR, thanks all!

File: sound/soc/sof/xtensa/core.c [hunk 1]
```diff
@@ -109,36 +109,19 @@ static void xtensa_stack(struct snd_sof_dev *sdev, void *oops, u32 *stack,
...
-		break;
-	case 3:
-		dev_err(sdev->dev, "0x%8.8x: 0x%8.8x 0x%8.8x 0x%8.8x\n",
-			stack_ptr + stack_words - 3, stack[stack_words - 3],
-			stack[stack_words - 2], stack[stack_words - 1]);
-		break;
-	default:
-		break;
+	/* example output:
+	 * 0x0049fbb0: 8000f2d0 0049fc00 6f6c6c61 00632e63
```
Comment (plbossart) [inline sound/soc/sof/xtensa/core.c:119]: that looks different from the previous format? Does this work with the tools?

Comment (RanderWang) [reply sound/soc/sof/xtensa/core.c:119]: yes, this is the result of hex_dump_to_buffer, which makes a little different format from our output. Now, for we have spent time to make it clear, we have to refine our tools according to this change.

Comment (plbossart) [reply sound/soc/sof/xtensa/core.c:119]: so if I merge this today it breaks the tools.  Not going to happen until you provide the matching change in the tools.

Comment (RanderWang) [reply sound/soc/sof/xtensa/core.c:119]: @plbossart  The tools is coredumper in our tools directory. I check it and find that it has dumped stack from memory windows of dsp now, not from kernel message which is also gotten from memory window (@mwierzbix is the maintainer). So this patch doesn't affect the tool now.

Comment (plbossart) [reply sound/soc/sof/xtensa/core.c:119]: ok thanks

File: sound/soc/sof/xtensa/core.c [hunk 2]
```diff
@@ -109,36 +109,19 @@ static void xtensa_stack(struct snd_sof_dev *sdev, void *oops, u32 *stack,
 {
 	struct sof_ipc_dsp_oops_xtensa *xoops = oops;
 	u32 stack_ptr = xoops->stack;
+	/* 4 dwords + 3 spaces + 1 terminating NUL */
+	unsigned char buf[8 * 4 + 3 + 1];
```
Comment (plbossart) [inline sound/soc/sof/xtensa/core.c]: one doubt here: is this really 8 bytes per value? You seem to use 16 + 3 + 1 with the format you selected:
0x0049fbb0: 8000f2d0 0049fc00 6f6c6c61 00632e63

Something's not adding up, not sure what I am missing or if this is a mistake

Comment (RanderWang) [reply sound/soc/sof/xtensa/core.c]: 0x0049fbb0: 8000f2d0 0049fc00 6f6c6c61 00632e63, here "0x0049fbb0: " is not generated by hex_dump_to_buffer, it is by "dev_err(sdev->dev, "0x%8.8x:".   Only "8000f2d0 0049fc00 6f6c6c61 00632e63" is generated by hex_dump_to_buffer.  Each dword is convert to ASCII of 8 char.

Comment (RanderWang) [reply sound/soc/sof/xtensa/core.c]: hex_dump_to_buffer does't create prefix of address.

**Comment by RanderWang** [comment]
update my comment, thanks!

**Pull request merged**